pkg/machine/machinesetlist.go

```diff
@@ -0,0 +1,49 @@
+package machine
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/golang/glog"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ListWorkerMachineSets returns the list of slice of MachineSetBuilder objects in a namespace on a cluster.
+func ListWorkerMachineSets(apiClient *clients.Settings, namespace string) ([]*MachineSetBuilder, error) {
+	if namespace == "" {
+		glog.V(100).Infof("machineSet 'namespace' parameter can not be empty")
+
+		return nil, fmt.Errorf("failed to list MachineSets, 'namespace' parameter is empty")
+	}
+
+	machineSetList, err := apiClient.MachineSets(namespace).List(context.TODO(), metav1.ListOptions{})
+
+	if err != nil {
+		glog.V(100).Infof("Failed to list MachineSets in the namespace %s due to %s",
+			namespace, err.Error())
+
+		return nil, err
+	}
+
+	var (
+		machineSetObjects     []*MachineSetBuilder
+		workerMachineSetLabel = "machine.openshift.io/cluster-api-machine-role"
```

Comment:
May be better to move it in to func parameter instead of hard-coding this var?

Reply:
I set it as a func param in machinesetlist.go and a var in machineset.go.  It is a label specific to machineset and trying not to have to to set it and pass this label from the test case since it is specific to the machineset object.

Reply:
What if developer change this label tomorrow? I believe that eco-goinfra should be generic enough to survive default value/labels changes.

Reply:
Ok, I'll set the workerLabel as a func argument, we can set it in the test case.